jdaviz/configs/imviz/tests/test_delete_data.py

```diff
@@ -0,0 +1,67 @@
+import numpy as np
+from astropy.io import fits
+from numpy.testing import assert_allclose
+from regions import PixCoord, CirclePixelRegion, RectanglePixelRegion
+
+from jdaviz.configs.imviz.tests.utils import BaseImviz_WCS_WCS
+
+
+class TestDeleteData(BaseImviz_WCS_WCS):
+
+    def test_delete_with_subset_wcs(self):
+        # Add a third dataset to test relinking
+        arr = np.ones((10, 10))
+
+        # First data with WCS, same as the one in BaseImviz_WCS_NoWCS.
+        hdu3 = fits.ImageHDU(arr, name='SCI')
+        hdu3.header.update({'CTYPE1': 'RA---TAN',
+                            'CUNIT1': 'deg',
+                            'CDELT1': -0.0002777777778,
+                            'CRPIX1': 1,
+                            'CRVAL1': 337.5202808,
+                            'NAXIS1': 10,
+                            'CTYPE2': 'DEC--TAN',
+                            'CUNIT2': 'deg',
+                            'CDELT2': 0.0002777777778,
+                            'CRPIX2': 1,
+                            'CRVAL2': -20.833333059999998,
+                            'NAXIS2': 10})
+        self.imviz.load_data(hdu3, data_label='has_wcs_3')
+
+        self.imviz.link_data(link_type='wcs', wcs_fallback_scheme=None, error_on_fail=True)
+
+        # Add a subset
+        reg = CirclePixelRegion(PixCoord(2, 2), 3)
+        self.imviz.load_regions(reg)
+
+        reg = RectanglePixelRegion(PixCoord(1, 1), 2, 2)
+        self.imviz.load_regions(reg)
+
+        assert len(self.imviz.app.data_collection.subset_groups) == 2
+
+        subset1 = self.imviz.app.data_collection.subset_groups[0]
+        subset2 = self.imviz.app.data_collection.subset_groups[1]
+        assert subset1.subset_state.xatt.parent.label == "has_wcs_1[SCI,1]"
+        assert_allclose(subset1.subset_state.center(), (2, 2))
+
+        assert subset2.subset_state.xatt.parent.label == "has_wcs_1[SCI,1]"
+        assert_allclose(subset2.subset_state.roi.xmin, 0)
+        assert_allclose(subset2.subset_state.roi.ymin, 0)
+        assert_allclose(subset2.subset_state.roi.xmax, 2)
+        assert_allclose(subset2.subset_state.roi.ymax, 2)
+
+        self.imviz.app.remove_data_from_viewer("imviz-0", "has_wcs_1[SCI,1]")
+        self.imviz.app.vue_data_item_remove({"item_name": "has_wcs_1[SCI,1]"})
```

Comment:
Hmm, so `remove_data_from_viewer` does not update the data menu automatically? So if a dev wants to remove a data via API, do they always have to call this pair of commands in this order?

Reply:
The names here are super confusing and should probably be changed now that you mention it (I didn't make them 😅). `remove_data_from_viewer` unloads the data from the viewer and is called by `vue_data_item_unload` (the `X` button), whereas `vue_data_item_remove` corresponds to the trashcan button. I should probably call `vue_data_item_unload` here instead of `remove_data_from_viewer` to be consistent.

Reply:
Ah, right, so you cannot actually delete the data until it is unloaded? Should probably comment on that here, since it is very subtle.